### ISSUE TITLE
deprecated cross product

### ISSUE THREAD

**Issue by RiccardoRossi** [KratosMultiphysics/Kratos#1186]
@loumalouomega 

doing a blame it looks like that you commented on the line

https://github.com/KratosMultiphysics/Kratos/blame/master/applications/StructuralMechanicsApplication/custom_elements/shell_thin_element_3D4N.cpp#L1856

this IS deprecated and it will soon fail to compile. The corresponding code should be changed from

       			const Vector vec1 = Vector(MathUtils<double>::CrossProduct(r_13*-1.0, e_24)); // NOTE: This operation is marked as deprecated by the compiler, but it is not deprecated...not my fault

to (at least as i understand...)

      Vector vec1(3);
      MathUtils<double>::CrossProduct(vec1, r_13, e_24)

where what i did is to take out the (-1.0)

the reason for which it was failing is that you were passing an expression (the multiplication by 1.0)

**Comment by loumalouomega** [comment]
I commented in that PR that for some reason the tests failed if I change
the cross product. It is not my code, so I prefer to delegate the
correction to the element creator.

**Comment by loumalouomega** [comment]
From #1051 

![imagen](https://user-images.githubusercontent.com/19991680/34079063-e78eb596-e325-11e7-813c-a526ef115271.png)

**Comment by RiccardoRossi** [comment]
@KratosMultiphysics/structural-mechanics who implemented this shell?

**Comment by philbucher** [comment]
@AndreasWinterstein and my Student implemented that

I am anyway working on this now, @RiccardoRossi can you pls tell me what to do?
Or can you fix it?

**Comment by roigcarlo** [comment]
At @KratosMultiphysics/technical-committee  we have tried to compile the code and it gives no problems:

We have changed:
```C++
const Vector vec1 = Vector(MathUtils<double>::CrossProduct(r_13*-1.0, e_24));
```

For:
```C++
Vector vec1 = ZeroVector(3);
MathUtils<double>::CrossProduct(vec1, r_13, e_24);
```

Which should solve both the deprecated problem and the problem with the sign, as the normal in the new function gives the correct sign

**Comment by maceligueta** [comment]
So, if this solution looks fine, a PR can be created with this change. @philbucher

**Comment by RiccardoRossi** [comment]
minor: you don't need the zero initialization

**Comment by maceligueta** [comment]
So, Vector is resized automatically?

**Comment by jcotela** [comment]
No, but CrossProduct will try to resize if not the right size (I just checked). I think this is a bad idea, however, since this will produce compilation errors if we try call CrossProduct for fixed-size types that don't implement a resize method.

**Comment by philbucher** [comment]
Guys I am still confused
Is this what I should use now?
```
Vector vec1 = ector(3);
MathUtils<double>::CrossProduct(vec1, r_13, e_24);
```

**Comment by jcotela** [comment]
@philbucher Yes, but please check that we didn't change the sign of the result: now we should be doing vec1 = r_13 x e_24, while it originally was vec1 = e_24 x (-r_13).

**Comment by philbucher** [comment]
fixed in #1231
